### PR TITLE
Delay IOMMU Install until DMAR is available

### PR DESCRIPTION
**Pull request by Javagedes**
## Description

Delay IOMMU Install until DMAR is available

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

202311

## Integration Instructions

N/A

### CODE REVIEW

File: IntelSiliconPkg/Feature/VTd/IntelVTdDxe/DmaProtection.c [hunk 1]
```diff
@@ -575,7 +575,8 @@ AcpiNotificationFunc (
   IN VOID       *Context
   )
 {
-  EFI_STATUS  Status;
+  EFI_STATUS          Status;
+  EFI_HANDLE          Handle;  //MU_CHANGE
```
Comment (os-d) [inline IntelSiliconPkg/Feature/VTd/IntelVTdDxe/DmaProtection.c]: nit: The old spacing was fine, no need to change Status here and Handle can sit and same spacing.

File: IntelSiliconPkg/Feature/VTd/IntelVTdDxe/DmaProtection.h [hunk 2]
```diff
@@ -137,7 +137,10 @@ extern VTD_UNIT_INFORMATION  *mVtdUnitInformation;
 extern UINT64  mBelow4GMemoryLimit;
 extern UINT64  mAbove4GMemoryLimit;
 
-extern EDKII_PLATFORM_VTD_POLICY_PROTOCOL  *mPlatformVTdPolicy;
+extern EDKII_PLATFORM_VTD_POLICY_PROTOCOL   *mPlatformVTdPolicy;
```
Comment (os-d) [inline IntelSiliconPkg/Feature/VTd/IntelVTdDxe/DmaProtection.h]: nit: this line didn't need to change

**Review by os-d — APPROVED** [review]

File: IntelSiliconPkg/Feature/VTd/IntelVTdDxe/DmaProtection.h (hunk 2 above)
Comment (Javagedes) [reply IntelSiliconPkg/Feature/VTd/IntelVTdDxe/DmaProtection.h]: Adjusted! Thanks

File: IntelSiliconPkg/Feature/VTd/IntelVTdDxe/DmaProtection.c (hunk 1 above)
Comment (Javagedes) [reply IntelSiliconPkg/Feature/VTd/IntelVTdDxe/DmaProtection.c]: Adjusted. Thanks!

**Review by apop5 — APPROVED** [review]

**Review by makubacki — APPROVED** [review]

**Pull request merged**